software/spmd/bsg_cuda_lite_runtime/barrier/Makefile

```diff
@@ -20,7 +20,7 @@ all: main.run
 
 KERNEL_NAME ?=kernel_barrier
 
-OBJECT_FILES=main.o bsg_set_tile_x_y.o bsg_printf.o kernel_barrier.o
+OBJECT_FILES=main.o bsg_tile_config_vars.o bsg_set_tile_x_y.o bsg_printf.o kernel_barrier.o
```

Comment:
maybe a separate list for common objs and reuse them?

Reply:
Ideally, we'd re-use the .a file that is generated. But that's not part of this PR. Perhaps @vb000 can comment, or add it in the linker script refactor?

Reply:
(i.e, yes I agree with you :p )

Reply:
Right, I totally missed that! We use a make variable called `$(SPMD_COMMON_OBJECTS)` that does that. Sure, I could do that.

Reply:
Why not link with the .a file, though? This works in baseline...

Reply:
And we do it in SPMD

Reply:
We have `.a` as well. SPMD_COMMON_OBJECTS was the old way of doing it. I think `.a` should supersede it.

Reply:
Perfect! This all makes sense now

Reply:
Filed an issue